src/services/projects.js

```diff
@@ -260,4 +264,132 @@ module.exports = class ProjectsHelper {
 			throw error
 		}
 	}
+
+	static async submitForReview(userDetails, resourceId, bodyData) {
+		try {
+			let projectDetails = await this.details(resourceId, userDetails.organization_id, userDetails.id)
+
+			if (projectDetails.statusCode == httpStatusCode.ok) {
+				let projectData = projectDetails.result
+				if (projectData.type !== common.PROJECT) {
+					return responses.failureResponse({
+						message: 'ONLY_PROJECT_CAN_BE_SUBMITTED_FOR_REVIEW',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+						result: utils.errorObject('body', 'type'),
+					})
+				}
+				if (projectData.title == '' || projectData.title == null) {
+					return responses.failureResponse({
+						message: 'PROJECT_TITLE_NOT_ADDED',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+						result: utils.errorObject('body', 'title'),
+					})
+				}
+				if (projectData.objective == '') {
+					return responses.failureResponse({
+						message: 'PROJECT_OBJECTIVE_NOT_ADDED',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+						error: utils.errorObject('body', 'objective'),
+					})
+				}
+				if (projectData.keywords == '') {
+					return responses.failureResponse({
+						message: 'PROJECT_KEYWORD_NOT_ADDED',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+						error: utils.errorObject('body', 'keywords'),
+					})
+				}
+				let entitiyTypes = await entityModelMappingQuery.findModelAndEntityTypes({
+					model: common.PROJECT,
+					status: common.STATUS_ACTIVE,
+				})
+				let entities = []
+				for (let i = 0; i < entitiyTypes.length; i++) {
+					if (projectData[entitiyTypes[i]] && projectData[entitiyTypes[i]] != '') {
+						entities.push(projectData[entitiyTypes[i]].id)
+					}
+				}
+				let allEntities = await entityService.read({ id: entities }, userDetails.id)
+				for (let i = 0; i < entitiyTypes.length; i++) {
+					let entitiesPresent = allEntities.result.find(
+						(item) => item.value === projectData[entitiyTypes[i]].value
+					)
+					if (!entitiesPresent) {
+						return responses.failureResponse({
+							message: entitiyTypes[i] + ' not added',
+							statusCode: httpStatusCode.bad_request,
+							responseCode: 'CLIENT_ERROR',
+							error: utils.errorObject('body', entitiyTypes[i]),
+						})
+					}
+				}
+				if (projectData.tasks.length < 1) {
```

Comment:
I think we should have entity types for project task and based on the validations we can put it here.